doc/Data Plane Code Generation/ServiceDrivenEvolution.md

```diff
@@ -0,0 +1,177 @@
+# Service driven evolution
+
+As a service evolves with non-breaking changes in the REST API, it may or may not cause breaking changes in the generated SDK. For instance, adding a new method in an existing path or in a new path won't cause breaking changes in the new SDK. But some changes like adding an optional query parameter could cause breaking changes in the generated SDK. In order to make the SDK backward compatible, we can add customized code such as overload methods. The following sections will show some examples of such scenarios and how to handle them.
+
+## A method gets a new optional parameter
+
+Generated code in a V1 client with a required query parameter and an optional query parameter:
+
+``` C#
+public virtual async Task<Response> PutRequiredOptionalAsync(string requiredParam, string optionalParam = null, RequestContext context = null)
+{
+    Argument.AssertNotNull(requiredParam, nameof(requiredParam));
```

Comment:
We should not be validating query parameters in our clients.
https://azure.github.io/azure-sdk/dotnet_introduction.html#parameter-validation

Reply:
Just curious: what is the difference between URL segments and path parameters. It says we should validate path parameter, while should not validate URL segments.

Reply:
@AlexanderSher, is this a bug in the guidelines?